content.en/tools/tiros.md

```diff
@@ -0,0 +1,230 @@
+# Tiros
+
+Tiros is an IPFS website measurement tool. It orchestrates the interplay between
+a Kubo node and a headless Chrome instance. The goal is to measure website
+metrics like [TTFB](https://web.dev/ttfb) or [FCP](https://web.dev/fcp) when
+loaded over IPFS and compare them with their HTTPS performance equivalents.
+For that, tiros instructs Chrome to request a website through the gateway
+of a local Kubo node or regularly over HTTPS.
+
+{{< button href="https://github.com/dennis-tra/nebula" >}}GitHub{{< /button >}}
+
+## Deployment
+
+We are running Tiros on ECS as a scheduled task every six hours in the following
+seven AWS regions:
+
+```
+eu-central-1
+ap-south-1
+af-southeast-2
+sa-east-1
+us-east-2
+us-west-1
+af-south-1
+```
+
+The list of websites that we are currently monitoring is configured [here](https://github.com/protocol/probelab-infra/blob/aabe20d28e833364e0ed17e651d5f810e524cbb9/aws/tf/modules/tiros/_variables.tf#L49).
+At the time of writing the list is:
+
+```
+filecoin.io
+libp2p.io
+ipld.io
+protocol.ai
+docs.ipfs.tech
+tldr.filecoin.io
+green.filecoin.io
+drand.love
+web3.storage
+consensuslab.world
+strn.network
+research.protocol.ai
+ipfs.tech
+blog.ipfs.tech
+docs.libp2p.io
+blog.libp2p.io
+en.wikipedia-on-ipfs.org/wiki
+```
+
+Each ECS task consists of three containers:
+
+1. `scheduler` - [tiros repository](https://github.com/plprobelab/tiros)
+2. `chrome` - via [`browserless/chrome`](https://github.com/browserless/chrome)
+3. `kubo` - via [ipfs/kubo](https://hub.docker.com/r/ipfs/kubo/)
+
+`kubo` is running with `LIBP2P_RCMGR=0` which disables the [libp2p Network Resource Manager](https://github.com/libp2p/go-libp2p-resource-manager#readme).
+
+The `scheduler` gets configured with a list of websites that will then be probed. A typical website config looks like this `ipfs.io,docs.libp2p.io,ipld.io`. The scheduler probes each website via `kubo` by requesting `http://localhost:8080/ipns/<website>` and via HTTP  by requesting`https://<website>`. Port `8080` is the default `kubo` HTTP-Gateway port. The `scheduler` uses [`go-rod`](https://github.com/go-rod/rod) to communicate with the `browserless/chrome` instance. The following excerpt is a gist of what's happening when requesting a website:
+
+```go
```

Comment:
Generally seems like a decent amount of code here and below we're copy/pasting in.  Can we just link to the actual source (since this will get out of date)?

Reply:
Yeah, that's indeed not nice here. I removed all the code bits.